src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -393,7 +452,317 @@ public virtual bool TryInsertExpansion(int startPositionInSubjectBuffer, int end
+                    new XElement(snippetNamespace + "Default", new XText("")),
+                    new XElement(snippetNamespace + "Function")));
+            }
+
+            if (includeMethod)
+            {
+                template.Append(")$end$");
+            }
+
+            // A snippet is manually constructed. Replacement fields are added for each argument, and the field name
+            // matches the parameter name.
+            // https://docs.microsoft.com/en-us/visualstudio/ide/code-snippets-schema-reference?view=vs-2019
+            return new XDocument(
+                new XDeclaration("1.0", "utf-8", null),
+                new XElement(
+                    snippetNamespace + "CodeSnippets",
+                    new XElement(
+                        snippetNamespace + "CodeSnippet",
+                        new XAttribute(snippetNamespace + "Format", "1.0.0"),
+                        new XElement(
+                            snippetNamespace + "Header",
+                            new XElement(
+                                snippetNamespace + "SnippetTypes",
+                                new XElement(snippetNamespace + "SnippetType", new XText("Expansion"))),
+                            new XElement(snippetNamespace + "Title", new XText(methodName)),
+                            new XElement(snippetNamespace + "Author", "Microsoft"),
+                            new XElement(snippetNamespace + "Description"),
+                            new XElement(snippetNamespace + "HelpUrl"),
+                            new XElement(snippetNamespace + "Shortcut", methodName)),
+                        new XElement(
+                            snippetNamespace + "Snippet",
+                            new XElement(snippetNamespace + "Declarations", declarations.ToArray()),
+                            new XElement(
+                                snippetNamespace + "Code",
+                                new XAttribute(snippetNamespace + "Language", "csharp"),
+                                new XCData(template.ToString()))))));
+        }
+
+        private void OnModelUpdated(object sender, Model e)
+        {
+            AssertIsForeground();
+
+            if (e is null)
+            {
+                return;
+            }
+
+            if (!_state.IsFullMethodCallSnippet)
+            {
+                return;
+            }
+
+            var document = SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document is null)
+            {
+                return;
+            }
+
+            var newSymbolKey = (e.SelectedItem as AbstractSignatureHelpProvider.SymbolKeySignatureHelpItem)?.SymbolKey ?? default;
+            var newSymbol = newSymbolKey.Resolve(document.Project.GetRequiredCompilationAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None), cancellationToken: CancellationToken.None).GetAnySymbol();
```

Comment:
this seems to contradict someting you wrote higher.  here we're doing sync blocking on async without JTF.  is that apropriate?  if it's appropriate here, why was it not appropriate above?